### PR TITLE
add kubernetes cronjob and logic

### PR DESCRIPTION
**Pull request by jterryons**
Add kubernetes cron job to handle the Sample service cron.
With Kubernetes cronjob concurrencyPolicy set to Forbid we wont have to worry about multiple crons running at the same time.

All redis lock gone. Redis removed from the pom.
cronjob schedule defaults to 5 mins but is configurable

Noise reduction:
- Removed swagger ui noise
- removed all SFTP config (theres no sftp code)
- Removed Jenkinsfile
- removed unused code from StateTransitionManager

Bumped minor patch

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -46,7 +46,6 @@ jobs:
 
       - name: Run Tests
         run: |
-          mvn fmt:check
```
Comment (insacuri) [inline .github/workflows/main.yml:49]: You sure we should be removing this?

Comment (mfcrocker) [reply .github/workflows/main.yml:49]: I would be very happy if we did

File: _infra/helm/sample/templates/cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    spec:
+      template:
+        spec:
+          containers:
+          - name: curl
+            image: curlimages/curl:7.71.0
+            command: ["/bin/sh", "-c"]
+            args:
+              - |
+                curl -X POST "http://{{ .Chart.Name }}:{{ .Values.dns.wellKnownPort }}/samples/export" -H "accept: application/json"
```
Comment (insacuri) [inline _infra/helm/sample/templates/cronjob.yaml:19]: One very small nit pick.  Can you add the -s and -S flags to this like in https://github.com/ONSdigital/rm-collection-exercise-service/blob/master/_infra/helm/collection-exercise/templates/distribute-sample-unit-cronjob.yaml#L37

The reason for this is that you get a bunch of 
```
0 0 0 0 0 0 0 0 --:--:-- --:--:-- --:--:-- 0 0 0 0 0 0 0 0 0 --:--:-- --:--:-- --:--:-- 0
```
and 
```
% Total % Received % Xferd Average Speed Time Time Time Current
```
 'error' logs when you don't suppress the output

**Review by insacuri — CHANGES_REQUESTED** [review]
One small comment, other then that it looks good.  Just testing it locally now

**Review by insacuri — APPROVED** [review]

**Pull request merged**